template/template.go

```diff
@@ -415,6 +476,38 @@ func (e *executer) resolveMatcher(m ast.Matcher) error {
 	return matcherErr
 }
 
+func (e *executer) hasAnnotation(operand ast.Value) bool {
```

Comment:
What happens in `.input {$n :number} .local $x = ${n} .match {$x} 1 {one} * {not one}`? `$x` has no annotation but it is a number.

Reply:
You already do it. Just noticed recursive call.